src/masternodes/masternodes.cpp

```diff
@@ -400,8 +400,11 @@ std::vector<int64_t> CMasternodesView::GetSubNodesBlockTime(const CKeyID & minte
     for (uint8_t i{0}; i < SUBNODE_COUNT; ++i) {
         ForEachSubNode([&](const SubNodeBlockTimeKey &key, int64_t blockTime)
         {
-            if (key.masternodeID == nodeId)
-            {
+            if (height >= Params().GetConsensus().FortCanningHeight) {
+                if (key.masternodeID == nodeId && key.subnode == i) {
+                    times[i] = blockTime;
+                }
+            } else if (key.masternodeID == nodeId) {
```

Comment:
I would prefer less code
```
auto subnode = height < Params().GetConsensus().FortCanningHeight || key.subnode == i;
if (key.masternodeID == nodeId && subnode)
```

Reply:
I hear you, but in this particular case, I'm inclined to preserve the clarity of the current code, branching on FC - much lesser cognitive effort to rationalize.